### ISSUE TITLE
What shared transformation type to use for flatfield correction

### ISSUE THREAD

**Issue by martinschorb** [saalfeldlab/render#114]
Hi,

I am trying to implement a flatfield correction using Render.

Would I need to specify the flatfield image as an array inside the JSON or can I provide a separate reference image somewhere on disk to be applied (multiply) to each tile? Also, does that array have to be full resolution, or can I provide a downscaled reference?

Thanks

**Comment by trautmane** [comment]
Hi Martin,

I assume you're looking for [flatfield correction like this](https://imagej.net/BigStitcher_Flatfield_correction).  Unfortunately, render does not currently support it.  I think you would need to do the correction upstream of render and use the corrected images as the source images in render.  Render does support multi-channel images and dynamic blending/rendering of multiple channels into a single image.  Maybe you could abuse that functionality to achieve the same effect - but I don't know if that would really work.

**Comment by martinschorb** [comment]
OK, thanks.

I thought that in the TEMCA data, you would also sometimes suffer from uneven illumination. Or do you correct for this during acquisition and tune the EM on the fly?

**Comment by trautmane** [comment]
You are correct, we did have uneven illumination in the TEMCA data.  This was addressed in 2 ways: (1) for point matching, we used a default filter that Saalfeld configured - this is what you see when using filter=true  and (2) after alignment, we used Misha Kazhdan's [DMG software](http://www.cs.jhu.edu/~misha/Code/DMG/Version4.51/) to perform 2D intensity correction for each layer.  Render also allows you to configure and name filters (or chains of filters) that can then be used during rendering.  Let me know if you'd like more information about that.
